hermit-abi/src/lib.rs

```diff
@@ -420,3 +428,36 @@ pub unsafe fn open(name: *const i8, flags: i32, mode: i32) -> i32 {
 pub unsafe fn unlink(name: *const i8) -> i32 {
 	sys_unlink(name)
 }
+
+/// The largest number `rand` will return
+pub const RAND_MAX: u64 = 2_147_483_647;
+
+/// The function computes a sequence of pseudo-random integers
+/// in the range of 0 to RAND_MAX
+#[inline(always)]
+pub unsafe fn rand() -> u32 {
+	sys_rand()
+}
+
+/// The function sets its argument as the seed for a new sequence
+/// of pseudo-random numbers to be returned by `rand`
+#[inline(always)]
+pub unsafe fn srand(seed: u32) {
+	sys_srand(seed);
+}
+
+/// Create a cryptographicly secure 32bit random number with the support of
+/// the underlying hardware. If the required hardware isn't available,
+/// the function returns `None`.
+#[inline(always)]
+pub unsafe fn secure_rand32() -> Option<u32> {
+	sys_secure_rand32()
+}
+
+/// Create a cryptographicly secure 64bit random number with the support of
+/// the underlying hardware. If the required hardware isn't available,
+/// the function returns `None`.
+#[inline(always)]
+pub unsafe fn secure_rand64() -> Option<u64> {
+	sys_secure_rand64()
+}
```

Comment:
Is there a reason these functions are unsafe? Will asking for random number ever invoke UB?

Reply:
I will answer the question in the comment below.